### ISSUE TITLE
Outdated warning doesn't meet a11y requirements

### ISSUE THREAD

**Issue by deniak** [w3c/tr-design#129]
@michael-n-cooper mentioned the new warning on outdated specs is not accessible.
Color contrast should be updated and keyboard support added.

cc @plehegar

**Comment by marcoscaceres** [comment]
Maybe worth considering using a dialog element and polyfill? https://github.com/GoogleChrome/dialog-polyfill

Together with a button or link, it would make this keyboard accessible and solve positioning issues.

**Comment by LJWatson** [comment]
As well as the keyboard accessibility problem (the expand/collapse control doesn't work), the warning is not being communicated to screen reader users (it's at the end of the document in the source order, regardless of how it's displayed visually).

@marcoscaceres' suggestion to use the ```<dialog>``` element would solve both the keyboard and screen reader cases, but we'd be using it in a slightly unusual way I think? Rather than a dialog that is either open or closed (and triggered by a button), we want a dialog that is either open (displayed prominently) or reduced (displayed discretely).

Providing keyboard focus was taken to the dialog and constrained within it whilst it was open, I think it would work though.

Suggest using a ```<button>``` for the control that opens/closes the dialog. The accessible name for the control also needs improving (ATM it's either "expand" or "collapse"). Suggest something like "Minimise warning" and "Maximise warning" - though if something else works better visually, a more explanatory accessible name can be provided using `aria-label` or somesuch.

**Comment by marcoscaceres** [comment]
Great suggestions, @LJWatson. Do also we need to (rudely?) announce the box being shown? It pops up asynchronously.

**Comment by LJWatson** [comment]
@marcoscaceres if/when the button in the footer is used to open the dialog, the same keyboard behaviour should be invoked I think. Is that what you meant?

**Comment by marcoscaceres** [comment]
> @marcoscaceres if/when the button in the footer is used to open the dialog, the same keyboard behaviour should be invoked I think. Is that what you meant?

Not exactly. What happens today is:

1. user types in URL into address bar.
1. The page loads (takes maybe 2-3 seconds).
1. The warning suddenly appears like a dialog: it literally takes over the display.
1. The user then has to make a choice to to minimize it - or they can navigate to the latest version via a provided link. 

Step 3 above, where the dialog suddenly appears, is quite abrupt (rude?) - and this is on purpose. So, I'm wondering, how should we announce the dialog being shown to the screen reader.

**Comment by LJWatson** [comment]
<Step 3 above, where the dialog suddenly appears, is quite abrupt (rude?) - and this is on purpose. So, I'm wondering, how should we announce the dialog
being shown to the screen reader.<

This is where grabbing keyboard focus and putting it on the ```<dialog>``` comes in. We'd also need to polyfill the accessibility semantics with something like:
```
<dialog role="alertdialog" aria-label="Warning">...</dialog>
```

If there's a visible piece of text that can be used as the accessible name for the dialog, then `aria-labelledby` could be used instead. The upshot is that when keyboard focus is moved to the ```<dialog>``` element, the screen reader should announce its role and name.

The next part is to make sure that keyboard focus stays within the dialog whilst it's open. This makes sure screen reader users and other keyboard users can't access the underlying page content until the dialog is dismissed.

**Comment by marcoscaceres** [comment]
@LJWatson thanks for the explanation! Makes sense.

**Comment by acli** [comment]
I just bumped into the warning “dialog” today (wanting to go to HTML 5.1 from the 5.2 spec (because I needed something published before December), not realizing it will go through every single draft).

What happens today is not “the user then has to make a choice to to minimize it - or they can navigate to the latest version via a provided link.” The user really doesn’t have a perceivable choice. The text is so dark it’s virtually black on black. The link to the previous version is obscured but still legible, but I didn’t even realize there was a button there until literally having gone through a dozen drafts and decided to just type in a guessed URL.

The wording is the other problem. I had no idea what “Collapse” might mean in the context. If it were labelled “Minimize” (and in white, so that I could actually see it) maybe I would have pressed it. I would have just gone to the debugger and deleted the dialog from the DOM rather than pressing a button with an unknown function.
